### ISSUE TITLE
Backups get triggered twice

### ISSUE THREAD

**Issue by adejanovski** [k8ssandra/k8ssandra#626]
## Bug Report

**Describe the bug**
Some backups are reported as failing although the Medusa container logs first say it succeeded but then complain about missing files:

```
[2021-04-02 11:52:06,140] INFO: Backup done
INFO:root:- Started: 2021-04-02 11:51:59
                        - Started extracting data: 2021-04-02 11:52:04
                        - Finished: 2021-04-02 11:52:06
[2021-04-02 11:52:06,140] INFO: - Started: 2021-04-02 11:51:59
                        - Started extracting data: 2021-04-02 11:52:04
                        - Finished: 2021-04-02 11:52:06
INFO:root:- Real duration: 0:00:02.088539 (excludes time waiting for other nodes)
[2021-04-02 11:52:06,140] INFO: - Real duration: 0:00:02.088539 (excludes time waiting for other nodes)
INFO:root:- 368 files, 714.22 KB
[2021-04-02 11:52:06,141] INFO: - 368 files, 714.22 KB
INFO:root:- 48 files copied from host
[2021-04-02 11:52:06,141] INFO: - 48 files copied from host
INFO:root:- 320 copied from previous backup (backup3)
[2021-04-02 11:52:06,141] INFO: - 320 copied from previous backup (backup3)
DEBUG:root:Emitting metrics
[2021-04-02 11:52:06,141] DEBUG: Emitting metrics
DEBUG:root:Done emitting metrics
[2021-04-02 11:52:06,141] DEBUG: Done emitting metrics
ERROR:root:This error happened during the backup: File /var/lib/cassandra/data/reaper_db/running_reapers-2c6d4800939d11eba8f41bcc7a5f0b4d/snapshots/medusa-backup4/md-5-big-CompressionInfo.db does not exist
[2021-04-02 11:52:25,977] ERROR: This error happened during the backup: File /var/lib/cassandra/data/reaper_db/running_reapers-2c6d4800939d11eba8f41bcc7a5f0b4d/snapshots/medusa-backup4/md-5-big-CompressionInfo.db does not exist
Traceback (most recent call last):
  File "/home/cassandra/medusa/backup_node.py", line 328, in backup_snapshots
    manifest_objects += storage.storage_driver.upload_blobs(src_batch, dst_path)
  File "/home/cassandra/medusa/storage/s3_base_storage.py", line 102, in upload_blobs
    multi_part_upload_threshold=int(self.config.multi_part_upload_threshold),
  File "/home/cassandra/medusa/storage/s3_compat_storage/concurrent.py", line 87, in upload_blobs
    return job.execute(list(src))
...
...
```

Scrolling up the logs we can see that the same backup was triggered twice in a row as the same sequence of log lines appear twice:

```
INFO:root:Performing backup backup4
[2021-04-02 11:51:59,822] INFO: Performing backup backup4
INFO:root:Monitoring provider is noop
[2021-04-02 11:51:59,822] INFO: Monitoring provider is noop
DEBUG:root:Loading storage_provider: s3_compatible
[2021-04-02 11:51:59,822] DEBUG: Loading storage_provider: s3_compatible
DEBUG:root:Reading credentials from /etc/medusa-secrets/medusa_s3_credentials
[2021-04-02 11:51:59,822] DEBUG: Reading credentials from /etc/medusa-secrets/medusa_s3_credentials
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): minio.minio.svc.cluster.local:9000
[2021-04-02 11:51:59,825] DEBUG: Starting new HTTP connection (1): minio.minio.svc.cluster.local:9000
DEBUG:urllib3.connectionpool:http://minio.minio.svc.cluster.local:9000 "HEAD /k8ssandra-medusa HTTP/1.1" 200 0
[2021-04-02 11:51:59,845] DEBUG: http://minio.minio.svc.cluster.local:9000 "HEAD /k8ssandra-medusa HTTP/1.1" 200 0
INFO:root:Performing backup backup4
[2021-04-02 11:51:59,866] INFO: Performing backup backup4
INFO:root:Monitoring provider is noop
[2021-04-02 11:51:59,866] INFO: Monitoring provider is noop
DEBUG:root:Loading storage_provider: s3_compatible
[2021-04-02 11:51:59,867] DEBUG: Loading storage_provider: s3_compatible
DEBUG:root:Reading credentials from /etc/medusa-secrets/medusa_s3_credentials
[2021-04-02 11:51:59,867] DEBUG: Reading credentials from /etc/medusa-secrets/medusa_s3_credentials
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): minio.minio.svc.cluster.local:9000
[2021-04-02 11:51:59,870] DEBUG: Starting new HTTP connection (1): minio.minio.svc.cluster.local:9000
DEBUG:urllib3.connectionpool:http://minio.minio.svc.cluster.local:9000 "HEAD /k8ssandra-medusa HTTP/1.1" 200 0
[2021-04-02 11:51:59,875] DEBUG: http://minio.minio.svc.cluster.local:9000 "HEAD /k8ssandra-medusa HTTP/1.1" 200 0
DEBUG:root:This server has systemd: False
[2021-04-02 11:52:02,421] DEBUG: This server has systemd: False
```

What explains the error is that when the first backup finishes, it clears the snapshot which then fails the second backup.

The error doesn't happen everytime since the previous backup (backup3) only triggered once:

```
INFO:root:Performing backup backup3
[2021-04-02 11:28:33,187] INFO: Performing backup backup3
INFO:root:Monitoring provider is noop
[2021-04-02 11:28:33,187] INFO: Monitoring provider is noop
DEBUG:root:Loading storage_provider: s3_compatible
[2021-04-02 11:28:33,188] DEBUG: Loading storage_provider: s3_compatible
DEBUG:root:Reading credentials from /etc/medusa-secrets/medusa_s3_credentials
[2021-04-02 11:28:33,188] DEBUG: Reading credentials from /etc/medusa-secrets/medusa_s3_credentials
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): minio.minio.svc.cluster.local:9000
[2021-04-02 11:28:33,190] DEBUG: Starting new HTTP connection (1): minio.minio.svc.cluster.local:9000
DEBUG:urllib3.connectionpool:http://minio.minio.svc.cluster.local:9000 "HEAD /k8ssandra-medusa HTTP/1.1" 200 0
[2021-04-02 11:28:33,211] DEBUG: http://minio.minio.svc.cluster.local:9000 "HEAD /k8ssandra-medusa HTTP/1.1" 200 0
DEBUG:root:This server has systemd: False
[2021-04-02 11:28:34,127] DEBUG: This server has systemd: False
WARNING:root:is ccm : 0
[2021-04-02 11:28:34,128] WARNING: is ccm : 0
DEBUG:root:Blob .k8ssandra/k8ssandra-dc1-default-sts-2.k8ssandra-dc1-service.k8ssandra.svc.cluster.local/backup3/meta/schema.cql was not found in cache.
[2021-04-02 11:28:34,147] DEBUG: Blob .k8ssandra/k8ssandra-dc1-default-sts-2.k8ssandra-dc1-service.k8ssandra.svc.cluster.local/backup3/meta/schema.cql was not found in cache.
DEBUG:root:[Storage] Getting object .k8ssandra/k8ssandra-dc1-default-sts-2.k8ssandra-dc1-service.k8ssandra.svc.cluster.local/backup3/meta/schema.cql
[2021-04-02 11:28:34,147] DEBUG: [Storage] Getting object .k8ssandra/k8ssandra-dc1-default-sts-2.k8ssandra-dc1-service.k8ssandra.svc.cluster.local/backup3/meta/schema.cql
DEBUG:urllib3.connectionpool:http://minio.minio.svc.cluster.local:9000 "HEAD /k8ssandra-medusa HTTP/1.1" 200 0
[2021-04-02 11:28:34,151] DEBUG: http://minio.minio.svc.cluster.local:9000 "HEAD /k8ssandra-medusa HTTP/1.1" 200 0
DEBUG:urllib3.connectionpool:http://minio.minio.svc.cluster.local:9000 "HEAD /k8ssandra-medusa/.k8ssandra/k8ssandra-dc1-default-sts-2.k8ssandra-dc1-service.k8ssandra.svc.cluster.local/backup3/meta/schema.cql HTTP/1.1" 404 0
[2021-04-02 11:28:34,155] DEBUG: http://minio.minio.svc.cluster.local:9000 "HEAD /k8ssandra-medusa/.k8ssandra/k8ssandra-dc1-default-sts-2.k8ssandra-dc1-service.k8ssandra.svc.cluster.local/backup3/meta/schema.cql HTTP/1.1" 404 0
DEBUG:root:Process psutil.Process(pid=1, name='python3', status='sleeping', started='10:17:48') was set to use only idle IO and CPU resources
[2021-04-02 11:28:34,156] DEBUG: Process psutil.Process(pid=1, name='python3', status='sleeping', started='10:17:48') was set to use only idle IO and CPU resources
INFO:root:Saving tokenmap and schema
[2021-04-02 11:28:34,156] INFO: Saving tokenmap and schema
DEBUG:cassandra.cluster:Connecting to cluster, contact points: ['10.8.2.194']; protocol version: 4
[2021-04-02 11:28:34,157] DEBUG: Connecting to cluster, contact points: ['10.8.2.194']; protocol version: 4
DEBUG:cassandra.pool:Host 10.8.2.194:9042 is now marked up
[2021-04-02 11:28:34,157] DEBUG: Host 10.8.2.194:9042 is now marked up
```

**To Reproduce**
1. Spin up a cluster with 3 Cassandra nodes
2. Load data using nosqlbench:
```
kubectl create job --image=nosqlbench/nosqlbench nosqlbench -n k8ssandra \
    -- java -jar nb.jar cql-iot rampup-cycles=1k cyclerate=50 \
    username=$username password=$password    \
    main-cycles=10k write_ratio=7 read_ratio=3 async=50       \
    hosts=k8ssandra-dc1-service --progress console:1s -v
```

3. Backup the cluster several times until it fails: `helm install my-backup charts/backup -n k8ssandra --set name=backup1,cassandraDatacenter.name=dc1` 
4. Check if the backup failed using `kubectl get cassandrabackup/<backup name> -o jsonpath={.status.failed}`
5. Check the logs of the failed node using `kubectl logs pod/<failed backup pod>` 

**Expected behavior**
Backups should be triggered only once per node.

**Environment (please complete the following information):**

* Helm charts version info

```
NAME      	NAMESPACE	REVISION	UPDATED                              	STATUS  	CHART                   	APP VERSION
k8ssandra 	k8ssandra	3       	2021-04-02 13:14:03.689639 +0200 CEST	deployed	k8ssandra-1.1.0-SNAPSHOT
minio     	minio    	1       	2021-04-02 11:26:38.048419 +0200 CEST	deployed	minio-8.0.10            	master
my-backup 	k8ssandra	1       	2021-04-02 13:22:55.207623 +0200 CEST	deployed	backup-0.26.0
my-backup2	k8ssandra	1       	2021-04-02 13:24:54.300259 +0200 CEST	deployed	backup-0.26.0
my-backup3	k8ssandra	1       	2021-04-02 13:28:32.02719 +0200 CEST 	deployed	backup-0.26.0
my-backup4	k8ssandra	1       	2021-04-02 13:51:58.485907 +0200 CEST	deployed	backup-0.26.0
traefik   	traefik  	1       	2021-04-02 11:34:41.413694 +0200 CEST	deployed	traefik-9.18.1          	2.4.8
```

* Helm charts user-supplied values

```
cassandra:
  datacenters:
  - name: dc1
    size: 3
  heap:
    newGenSize: 3G
    size: 8G
  ingress:
    enabled: false
  resources:
    limits:
      cpu: "8"
      memory: 16Gi
    requests:
      cpu: "8"
      memory: 16Gi
kube-prometheus-stack:
  enabled: false
  grafana:
    adminPassword: admin123
    adminUser: admin
    enabled: true
    grafana.ini:
      server:
        root_url: http://localhost:8080/grafana
        serve_from_sub_path: true
    ingress:
      enabled: false
      path: /grafana
  prometheus:
    enabled: true
    ingress:
      enabled: false
      paths:
      - /prometheus
    prometheusSpec:
      externalUrl: http://localhost:9090/prometheus
      routePrefix: /prometheus
medusa:
  bucketName: k8ssandra-medusa
  enabled: true
  image:
    repository: docker.io/jsanda/medusa
    tag: grpc-sigterm-13d70bb2
  multiTenant: true
  storage: s3_compatible
  storage_properties:
    host: minio.minio.svc.cluster.local
    port: 9000
    secure: "False"
  storageSecret: medusa-bucket-key
reaper:
  autoschedule: false
  enabled: true
  ingress:
    enabled: false
reaper-operator:
  enabled: true
stargate:
  cpuLimMillicores: 8000
  cpuReqMillicores: 4000
  enabled: true
  heapMB: 8096
  ingress:
    enabled: false
  replicas: 3
```

* Kubernetes version information:

```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-13T13:28:09Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.15-gke.1501", GitCommit:"de67bb0d58413ba2ba9b64810ab438a9734a2ab9", GitTreeState:"clean", BuildDate:"2021-02-26T23:41:52Z", GoVersion:"go1.13.15b4", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind:
GKE Autopilot

**Comment by jsanda** [comment]
I think this is a duplicate of https://github.com/k8ssandra/k8ssandra/issues/592.

**Comment by adejanovski** [comment]
I think so too, closing it.
